src/pages/correspondent-agent-branches/Windows/AgentBranchWindow.js

```diff
@@ -1,73 +1,71 @@
+    recordId: recordId || null,
+    agentBranch: null,
+    address: null,
+    addressId: null
+  })
+
+  const [activeTab, setActiveTab] = useState(0)
+  const tabs = [{ label: labels.agentBranch }, { label: labels.address, disabled: !store.recordId }]
+  const { postRequest } = useContext(RequestsContext)
+
+  async function onSubmit(address) {
+     if (!store.agentBranch.addressId) {
+      store.agentBranch.addressId = address.addressId
+      const res = { ...store.agentBranch, addressId: address.addressId }
+      if (res) {
+        const data = { ...res, recordId: store?.recordId, agentId: store.agentBranch?.agentId }
+        await postRequest({
+          extension: RemittanceSettingsRepository.CorrespondentAgentBranches.set,
+          record: JSON.stringify(data)
+        })
+      }
+    } 
+  }
+
+  function setAddress(res) {
+    setStore(prevStore => ({
+      ...prevStore,
+      address: res,
+      addressId: res.addressId
+    }))
+  }
 
-  tabs,
-  activeTab,
-  setActiveTab,
-  onSave,
-  agentBranchValidation,
-  labels,
-  agentStore,
-  setAgentStore,
-  fillStateStore,
-  fillCountryStore,
-  countryStore,
-  stateStore,
-  lookupCity,
-  cityStore,
-  setCityStore,
-  maxAccess,
-  lookupCityDistrict,
-  cityDistrictStore,
-  setCityDistrictStore
-}) => {
   return (
-    <Window
-      id='AgentWindow'
-      Title={labels.title}
-      tabs={tabs}
-      activeTab={activeTab}
-      setActiveTab={setActiveTab}
-      onClose={onClose}
-      width={width}
-      onSave={onSave}
-      agentBranchValidation={agentBranchValidation}
-    >
-      <CustomTabPanel index={0} value={activeTab}>
-        <AgentBranchTab
-          labels={labels}
-          setAgentStore={setAgentStore}
-          agentBranchValidation={agentBranchValidation}
+    <>
+      <CustomTabs tabs={tabs} activeTab={activeTab} setActiveTab={setActiveTab} />
+      <CustomTabPanel height={height} index={0} value={activeTab}>
+        <AgentBranchForm
+          _labels={labels}
           maxAccess={maxAccess}
-          agentStore={agentStore}
+          store={store}
+          setStore={setStore}
+          editMode={store.recordId}
         />
       </CustomTabPanel>
-
-      <CustomTabPanel index={1} value={activeTab}>
-        <AddressTab
-          countryStore={countryStore}
-          stateStore={stateStore}
-          labels={labels}
-          lookupCity={lookupCity}
-          fillStateStore={fillStateStore}
-          cityStore={cityStore}
-          setCityStore={setCityStore}
-          fillCountryStore={fillCountryStore}
-          addressValidation={agentBranchValidation}
+      <CustomTabPanel height={height} index={1} value={activeTab}>
+        <AddressForm
+          _labels={labels}
           maxAccess={maxAccess}
-          lookupCityDistrict={lookupCityDistrict}
-          cityDistrictStore={cityDistrictStore}
-          setCityDistrictStore={setCityDistrictStore}
+          editMode={store.recordId}
```

Comment:
Make it a const with value !!store.recordId

Reply:
okay but what is the difference
i remember that i was defining a const editMode= store.recordId
but a change was requested to use this directly